infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -393,10 +368,27 @@ private void applyShadowStream(UUID streamId, long snapshot) {
      * Read from shadowStream and append/apply to the actual stream
      */
     public void applyShadowStreams() {
-        log.debug("Apply Shadow Streams, total={}", streamViewMap.size());
-
+        log.debug("Apply Shadow Streams, total={}", config.getStreamMap().size());
         long snapshot = rt.getAddressSpaceView().getLogTail();
-        for (UUID regularStreamId : streamViewMap.keySet()) {
+
+        // Apply registry table first for streaming consideration in DYNAMIC mode.
+        if (config.getStreamMap().containsKey(REGISTRY_TABLE_ID)) {
+            applyShadowStream(REGISTRY_TABLE_ID, snapshot);
+            if (config.getStreamsDiscoveryMode().equals(StreamsDiscoveryMode.DYNAMIC)) {
+                // If LR is in DYNAMIC mode, signal the config to sync with registry table after applying its entries
+                config.syncWithRegistry();
+            }
+        } else {
+            // The only case this should happen is in some tests. Some error could occur if
```

Comment:
it is not expected even in tests.  Hitting it implies a bug or an incorrect test.  If you would like to check this condition, consider using a Precondition instead.

Reply:
this is not expected and tests should fail instead of logging it as an error.  Please add a precondition instead.

Reply:
Yes I totally agree, this should not even appear in tests, but actually we already have some LR ITs that manually tweak the workflow here, and registry table wasn't part of the streams to replicate in those tests. And I did feel modifying those tests is not worth it, we may have a discuss on how to deal with them...

Reply:
Please remove this case